Graphics/Vulkan/VulkanGraphicsMemoryAllocator.cpp

```diff
@@ -0,0 +1,159 @@
+                                          &outAllocation, &outAllocationInfo);
+
+        if (result != VK_SUCCESS)
+        {
+            throw Exceptions::InitialisationFailureException("Failed to correctly initialise the newly requested "
+                                                             "Vulkan Buffer based on the supplied createInfo object.",
+                                                             result);
+        }
+
+        return std::make_shared<VulkanGraphicsBuffer>(GetDevice(), shared_from_this(), createInfo.cpuAccessKind,
+                                                      createInfo.bufferKind, outAllocation, outAllocationInfo,
+                                                      outBuffer);
+    }
+
+    std::shared_ptr<VulkanGraphicsTexture> VulkanGraphicsMemoryAllocator::CreateVulkanTexture(
+        const GraphicsTextureCreateInfo& createInfo)
+    {
+        VkImageType imageType{};
+
+        switch (createInfo.textureKind)
+        {
+            case GraphicsTextureKind::OneDimensional:
+                imageType = VK_IMAGE_TYPE_1D;
+                break;
+            case GraphicsTextureKind::TwoDimensional:
+                imageType = VK_IMAGE_TYPE_2D;
+                break;
+            case GraphicsTextureKind::ThreeDimensional:
+                imageType = VK_IMAGE_TYPE_3D;
+                break;
+            default:
+            case GraphicsTextureKind::Unknown:
+                throw Exceptions::NotSupportedException(
+                    "The specified texture kind is not supported in the default Vulkan pipeline.");
+        }
+
+        VkExtent3D extent{};
+        extent.width = createInfo.width;
+        extent.height = createInfo.height;
+        extent.depth = createInfo.depth;
+
+        VkImageCreateInfo imageCreateInfo{};
+        imageCreateInfo.sType = VK_STRUCTURE_TYPE_IMAGE_CREATE_INFO;
+        imageCreateInfo.imageType = imageType;
+        imageCreateInfo.format = Utilities::Map(createInfo.texelFormat);
+        imageCreateInfo.extent = extent;
+        imageCreateInfo.mipLevels = 1;
+        imageCreateInfo.arrayLayers = 1;
+        imageCreateInfo.samples = VK_SAMPLE_COUNT_1_BIT;
+        imageCreateInfo.usage = Utilities::GetVulkanImageUsageKind(createInfo.textureKind, createInfo.gpuAccessKind);
+
+        VkImage vulkanImage = VK_NULL_HANDLE;
+        VmaAllocation outAllocation = VK_NULL_HANDLE;
+        VmaAllocationInfo outAllocationInfo{}; // TODO: I haven't figured out what I am doing with this yet, but I'm
+                                               // fairly sure we need it. Lol.
+
+        VmaAllocationCreateInfo allocationCreateInfo{};
+        allocationCreateInfo.flags = Utilities::GetVmaAllocationKind(createInfo.cpuAccessKind) | VMA_ALLOCATION_CREATE_DONT_BIND_BIT;
+        allocationCreateInfo.usage = VMA_MEMORY_USAGE_AUTO;
+        // allocationCreateInfo.priority = 1; //TODO: do I need this?
```

Comment:
You should need this if we need to prioritize the allocation over others. If everything is "de-prioritized", then no - simply don't set it or default it to 0.

Reply:
This is for future me to figure out I reckon, not for this PR.